### PR TITLE
configure login transport to allow for proxies

### PR DESCRIPTION
**Pull request by awsmsrc**
The implementation of retrieving the console user id in the login function broke the login command for users using proxies.

This should fix that.

### CODE REVIEW

**Comment by MiguelMoll** [comment]
Could we change the controller to return JSON instead of just OK? What would break? Out of scope?

**Comment by awsmsrc** [comment]
it shouldn't really break anything changing the controller to return json, as people won't be testing against that string, but login would still break with older racks

File: client/auth.go [hunk 1]
```diff
@@ -1,29 +1,43 @@
...
+	if err != nil {
+		return "", err
+	}
+
+	var id string
+	var data map[string]string
+
+	err = json.Unmarshal(body, &data)
+	if err == nil {
+		//if bad JSON is returned it is probably a legacy app
```
Comment (ddollar) [inline client/auth.go]: What does legacy app mean in this context?

Should probably document the "", nil potential return case here as it's out of the value-or-error normal.

File: client/auth.go [hunk 2]
```diff
@@ -1,29 +1,43 @@
...
 
 import (
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"net/http"
 )
 
-func (c *Client) Auth() error {
+func (c *Client) Auth() (string, error) {
```
Comment (ddollar) [inline client/auth.go]: Does /auth return more than just id in that JSON hash? If so this function should be returning a struct.

File: client/auth.go (hunk 1 above)
Comment (ddollar) [inline client/auth.go]: Does this "legacy app" situation happen enough to justify swallowing errors blindly here?

Comment (awsmsrc) [inline client/auth.go]: questions 1 + 3) app is meant to be rack, i.e anything before this release, will update the comment

all racks will return "", nil - that is the current released behaviour (user ids are a console concept), i will add a comment

**Comment by ddollar** [comment]
[convox login uses proxies so requires client code but still needs console id](https://trello.com/c/TqGSH7U2/2052-convox-login-uses-proxies-so-requires-client-code-but-still-needs-console-id)

**Comment by MiguelMoll** [comment]
https://github.com/convox/rack/pull/2006

**Pull request closed without merging**